src/PostgREST/SchemaCache.hs

```diff
@@ -614,7 +614,33 @@ tablesSqlQuery =
   -- on the "columns" CTE, left joining on pg_depend and pg_class is used to obtain the sequence name as a column default in case there are GENERATED .. AS IDENTITY,
   -- generated columns are only available from pg >= 10 but the query is agnostic to versions. dep.deptype = 'i' is done because there are other 'a' dependencies on PKs
   encodeUtf8 [trimming|
+   -- Recursively get the base types of domains
   WITH
+  base_types AS (
+    WITH RECURSIVE
+    recurse AS (
+      SELECT
+        oid,
+        typbasetype,
+        typnamespace AS base_namespace,
+        COALESCE(NULLIF(typbasetype, 0), oid) AS base_type
+      FROM pg_type
+      UNION
+      SELECT
+        t.oid,
+        b.typbasetype,
+        b.typnamespace AS base_namespace,
+        COALESCE(NULLIF(b.typbasetype, 0), b.oid) AS base_type
+      FROM recurse t
+      JOIN pg_type b ON t.typbasetype = b.oid
+    )
+    SELECT
+      oid,
+      base_namespace,
+      base_type
+    FROM recurse
+    WHERE typbasetype = 0
+  ),
```

Comment:
This should be deduplicated, it's the same as the below right?
https://github.com/PostgREST/postgrest/blob/deb11c3c2afc8dc19a630f6b8bfe1ed053c28ed2/src/PostgREST/SchemaCache.hs#L381-L403